### PR TITLE
fix(announce): Fixes links, mentions and hashtags for Bluesky.#1611

### LINKED ISSUE #1611: [announce] Links in Bluesky are not rendered correctly
Follow up to https://github.com/jreleaser/jreleaser/issues/1525

JReleaser 1.11.0 rendered links without format

https://bsky.app/profile/jreleaser.bsky.social/post/3kmkmrzw56h24

jreleaser-bluesky

### PR DESCRIPTION
**Pull request by TomCools**
Fixes #1611

### Context
Fixes broken hyperlinks, mentions and hashtags.

I have tagged the Jreleaser Bluesky account in some test posts, so you can verify it works.

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/jreleaser/jreleaser/blob/master/CONTRIBUTING.adoc).
- [ ] Make sure all contributed code can be distributed under the terms of the 
      [Apache License 2.0](https://github.com/jreleaser/jreleaser/blob/master/LICENSE), e.g. the code was written by 
      you or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Update [documentation](https://github.com/jreleaser/jreleaser.github.io) when applicable.

### CODE REVIEW

**Comment by TomCools** [comment]
The BlueskyStringEncodingWrapper was inspired by the following answer on SO: https://stackoverflow.com/a/58553748. Checking with you @aalmiray, should we attribute this persons answer in the codebase?

File: sdks/jreleaser-bluesky-java-sdk/src/main/java/org/jreleaser/sdk/bluesky/BlueskyRecordFactory.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+ * @since 1.12.0
+ */
+class BlueskyRecordFactory {
+
+    private static final Pattern URL_PATTERN = Pattern.compile(
+        "(?:^|[\\W])((ht|f)tp(s?):\\/\\/|www\\.)"
+            + "(([\\w\\-]+\\.){1,}?([\\w\\-.~]+\\/?)*"
+            + "[\\p{Alnum}.,%_=?&#\\-+()\\[\\]\\*$~@!:/{};']*)",
+        Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL);
+    private static final Pattern TAG_PATTERN = Pattern.compile("(#[a-zA-z]{2,})");
```
[the remark this answers is not in the document]
Comment (aalmiray) [reply sdks/jreleaser-bluesky-java-sdk/src/main/java/org/jreleaser/sdk/bluesky/BlueskyRecordFactory.java:57]: @TomCools looks like this should be `a-zA-Z` (capitalized Z)

**Review by aalmiray — CHANGES_REQUESTED** [review]
A regex needs update

**Comment by aalmiray** [comment]
We would have to verify the code is his own first 🤷‍♂️

**Comment by TomCools** [comment]
I was inspired by his code.. but the idea to "implement a new class of CharSequence and pass that to Matcher for regex matching" only can have so much variations.

I don't think checking his code ownership is feasible... how do we proceed from here? (Releasing 1.12 without this fix is an option ofc)

**Pull request merged**